platform/FEATURE_EXPERIMENTAL_API/FEATURE_PSA/src/mbedtls_crypto_setup.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2021 Arm Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "crypto_device_platform.h"
+#include "psa/crypto.h"
+
+// Required by mbedtls_platform_setup()
+
+int crypto_platform_setup(crypto_platform_ctx *unused)
```

Comment:
Should we have an ifdef to catch if cryptocell is enabled to provide a more friendly error message than duplicate symbol definition? We can't have both PSA and cryptocell enabled, unless we chain together their initializers.

Reply:
Good point, I need to check whether Mbed TLS can use CryptoCell and PSA Crypto at the same time. If it does, we need to make sure they are both initialised.

Reply:
It's tricky to initialise both CryptoCell and PSA Crypto here, because any crypto hardware (e.g. CryptoCell) relies on `MBEDTLS_PLATFORM_SETUP_TEARDOWN_ALT` (from Mbed TLS) to add initialisation hooks, but the same macro is for overriding `mbedtls_platform_setup()` for PSA Crypto too. So, there's no way to check if only PSA is available or PSA and some crypto hardware both need to be initialised - we just don't have any macro or anything for this.

Introducing a new config/macro would be a breaking change - it's easy to change CryptoCell as it's in Mbed OS, but if a third-party project has its own crypto hardware integrated with the same hook, it will break. Breaking changes (without a backward compatibility mechanism) are only allowed in Mbed OS between major releases (e.g. 5 -> 6).

So I just've completely changed the approach, and simply call `psa_crypto_init()` in `TLSSocketWrapper` if `MBEDTLS_USE_PSA_CRYPTO`. This is what [an example program of Mbed TLS](https://github.com/ARMmbed/mbedtls/blob/8682faeb092f32020602514fb0374d2ad1ce541a/programs/ssl/ssl_client2.c#L739-L748) does too.